react/features/app-settings/components/BackButton.native.js

```diff
@@ -0,0 +1,56 @@
+// @flow
+
+import React, { Component } from 'react';
+import { TouchableOpacity } from 'react-native';
+
+import styles from './styles';
+
+import { Icon } from '../../base/font-icons';
+import { Platform } from '../../base/react';
+
+/**
+* The type of the React {@code Component} props of {@link BackButton}
+*/
+type Props = {
+
+    /**
+    * The action to be performed when the button is pressed.
+    */
+    onPress: Function,
+
+    /**
+    * An external style object passed to the component.
+    */
+    style: Object
+};
+
+/**
+ * A component rendering a back button that looks native on both platforms.
+ */
+export default class BackButton extends Component<Props> {
+
+    /**
+     * Implements React's {@link Component#render()}, renders the button.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const isAndroid = Platform.OS === 'android';
+
+        return (
+            <TouchableOpacity
+                accessibilityLabel = { 'Back' }
+                onPress = { this.props.onPress }>
+                <Icon
+                    name = {
+                        isAndroid ? 'arrow_back' : 'navigate_before'
```

Comment:
extract this computation to a constant at the top level, there is no reason to compute it every time